interpreter/shared/src/test/scala/sigma/serialization/DataSerializerSpecification.scala

```diff
@@ -132,6 +132,20 @@ class DataSerializerSpecification extends SerializationSpecification {
       t.isInstanceOf[SerializerException] &&
           t.getMessage.contains(s"BigInt value doesn't not fit into ${SBigInt.MaxSizeInBytes} bytes")
     })
+  }
+
+  property("header roundtrip") {
+    VersionContext.withVersions(VersionContext.V6SoftForkVersion, 1) {
+      forAll { x: Header => roundtrip[SHeader.type](x, SHeader) }
+    }
 
+    an[SerializerException] should be thrownBy (
```

Comment:
For negative cases there is `assertExceptionThrown` which allows testing not only exception type but also the message. See other usages of it as an example.